### PR TITLE
Trash functionality

### PR DESCRIPTION
**Pull request by sameep-s**
**Functionality added:**

- Move to Trash.
- Delete from trash.
- Restore from trash.

### CODE REVIEW

**Review by SaurabhChawda — APPROVED** [review]
Code looks fine to me but you can work on responsiveness.

**Review by JatuFaique — APPROVED** [review]
Code looks clean and implementation is clear.
Looks good to merge, however i noticed one empty file, please remove it before if not in use.

File: src/services/archive-services.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        };
+
+    } catch (e) {
+        console.error(e);
+    }
+
+}
+
+
+export async function moveToArchive(dispatch, token, noteId, note) {
```
Comment (bajajcodes) [inline src/services/archive-services.js:23]: No need of seaprate of props here, I guess.

**Review by bajajcodes — APPROVED** [review]
- Good use of note modal. 
(modal is nested everywhere, which I am not cofirm is good or bad)
- Code style is consistent.
- Div tag can be replaced with section tag.

Code is good to merge and features are working as expected.

**Pull request merged**